core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -444,12 +444,14 @@ public void renameTable(SessionContext context, TableIdentifier from, TableIdent
 
   @Override
   public boolean tableExists(SessionContext context, TableIdentifier identifier) {
-    Endpoint.check(endpoints, Endpoint.V1_TABLE_EXISTS);
-
     try {
       checkIdentifierIsValid(identifier);
-      client.head(paths.table(identifier), headers(context), ErrorHandlers.tableErrorHandler());
-      return true;
+      if (endpoints.contains(Endpoint.V1_TABLE_EXISTS)) {
```

Comment:
Not an issue of this PR, but I doubt that there is any Iceberg-REST-service implementation that does not implement the `HEAD` endpoint - simply because versions up to 1.7 did not fail.

I'd propose to replace the V1_TABLE/VIEW/NAMESPACE_EXISTS with V1_LOAD_TABLE/VIEW/NAMESPACE to revert to the pre-1.8 behavior.